apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx

```diff
@@ -94,6 +96,13 @@ const PremiumTextfield = (props: ICustomUsernameProps) => {
   const updateUsername = trpc.viewer.updateProfile.useMutation({
     onSuccess: async () => {
       onSuccessMutation && (await onSuccessMutation());
+      await update({
+        ...session,
+        user: {
+          ...session?.user,
+          username: inputUsernameValue,
+        },
+      });
```

Comment:
Same as my previous comment. Extra points for encapsulating this behavior in a custom `trpc.viewer.updateProfile.useMutation` hook and just use that.

```suggestion
      await update();
```

Reply:
LGTM